### PR TITLE
fixed edge cases with weird data + tested

### LINKED ISSUE #1543: Create QA test for different types of columns and data

### PR DESCRIPTION
**Pull request by JKL98ISR**
closes #1543

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -143,6 +148,88 @@ def fillna(data: pd.DataFrame):
     return train_ds, test_ds, clf
 
 
+@pytest.fixture(scope='session')
+def wierd_classification_dataset_and_model():
```
Comment (matanper) [inline tests/conftest.py:152]: I'd name it "KISS_weird_classification_data_and_model"

**Review by nirhutnik — APPROVED** [review]

File: deepchecks/core/check_utils/whole_dataset_drift_utils.py [hunk 2]
```diff
@@ -44,6 +44,8 @@ def run_whole_dataset_drift(train_dataframe: pd.DataFrame, test_dataframe: pd.Da
                             min_meaningful_drift_score: float,
                             with_display: bool):
     """Calculate whole dataset drift."""
+    numerical_features = np.array(numerical_features, dtype='object')
```
[the remark this answers is not in the document]
Comment (JKL98ISR) [reply deepchecks/core/check_utils/whole_dataset_drift_utils.py:47]: it's the feature names

File: deepchecks/core/check_utils/whole_dataset_drift_utils.py [hunk 3]
```diff
@@ -44,6 +44,8 @@ def run_whole_dataset_drift(train_dataframe: pd.DataFrame, test_dataframe: pd.Da
...
+    cat_features = np.array(cat_features, dtype='object')
```
[the remark this answers is not in the document]
Comment (JKL98ISR) [reply deepchecks/core/check_utils/whole_dataset_drift_utils.py:48]: cuz it's any hashable,
and no, it just won't support any feature that aren't str in the fit

File: deepchecks/core/check_utils/whole_dataset_drift_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nirhutnik) [reply deepchecks/core/check_utils/whole_dataset_drift_utils.py:47]: Sorry, misread the line

**Review by matanper — APPROVED** [review]
